flink-batch-connectors/flink-avro/src/test/java/org/apache/flink/api/java/io/AvroOutputFormatTest.java

```diff
@@ -0,0 +1,126 @@
+package org.apache.flink.api.java.io;
+
+import static org.apache.flink.api.java.io.AvroOutputFormat.Codec;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.apache.flink.api.io.avro.example.User;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.junit.Test;
+
+/**
+ * Tests for {@link AvroOutputFormat}
+ */
+public class AvroOutputFormatTest {
+
+    @Test
+    public void testSetCodec() throws Exception {
+        // given
+        final AvroOutputFormat<User> outputFormat = new AvroOutputFormat<>(User.class);
+
+        // when
+        try {
+            outputFormat.setCodec(Codec.SNAPPY);
+        } catch (Exception ex) {
+            // then
+            fail("unexpected exception");
+        }
+    }
+
+    @Test
+    public void testSetCodecError() throws Exception {
+        // given
+        boolean error = false;
+        final AvroOutputFormat<User> outputFormat = new AvroOutputFormat<>(User.class);
+
+        // when
+        try {
+            outputFormat.setCodec(null);
+        } catch (Exception ex) {
+            error = true;
+        }
+
+        // then
+        assertTrue(error);
+    }
+
+    @Test
+    public void testSerialization() throws Exception {
+
+        serializeAndDeserialize(null);
+        for (final Codec codec : Codec.values()) {
+            serializeAndDeserialize(codec);
+        }
+    }
+
+    private void serializeAndDeserialize(final Codec codec) throws IOException, ClassNotFoundException {
+        // given
+        final AvroOutputFormat<User> outputFormat = new AvroOutputFormat<>(User.class);
+        if (codec != null) {
+            outputFormat.setCodec(codec);
+        }
+        outputFormat.setSchema(User.SCHEMA$);
+
+        final File serialized = File.createTempFile("avro-output-format", ".serialized");
+
+        // when
+        try (final ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(serialized))) {
```

Comment:
Please use a `ByteArrayOutputStream` instead of a `FileOutputStream`. That way the test is not touching the disk and therefore faster and more reliable.